ui/app/AppLayouts/CommunitiesPortal/popups/CreateCommunityPopup.qml

```diff
@@ -58,7 +59,278 @@ StatusStackModal {
+            }
+
+            ColumnLayout {
+                spacing: 12
+                visible: root.store.discordChannelsModel.count
+
+                StatusBaseText {
+                    Layout.fillWidth: true
+                    text: qsTr("Please select the categories and channels you would like to import")
+                    wrapMode: Text.WordWrap
+                }
+
+                RowLayout {
+                    spacing: 20
+                    Layout.fillWidth: true
+                    StatusRadioButton {
+                        text: qsTr("Import all history")
+                        checked: true
+                    }
+                    StatusRadioButton {
+                        id: startDateRadio
+                        text: qsTr("Start date")
+                    }
+                    StatusDatePicker {
+                        id: datePicker
+                        Layout.fillWidth: true
+                        selectedDate: new Date(root.store.discordOldestMessageTimestamp * 1000)
+                        enabled: startDateRadio.checked
+                    }
+                }
+
+                Rectangle {
+                    Layout.fillWidth: true
+                    Layout.fillHeight: true
+                    color: Theme.palette.baseColor4
+
+                    StatusListView {
+                        anchors.fill: parent
+                        anchors.margins: 16
+                        model: root.store.discordCategoriesModel
+                        delegate: ColumnLayout {
+                            width: ListView.view.width
+                            spacing: 8
+
+                            StatusCheckBox {
+                                readonly property string categoryId: model.id
+                                id: categoryCheckbox
+                                checked: model.selected
+                                text: model.name
+                                onToggled: checked ? root.store.selectDiscordCategory(categoryId) : root.store.unselectDiscordCategory(categoryId)
+                            }
+
+                            ColumnLayout {
+                                spacing: 8
+                                Layout.fillWidth: true
+                                Layout.leftMargin: 24
+                                Repeater {
+                                    Layout.fillWidth: true
+                                    model: root.store.discordChannelsModel
+                                    delegate: StatusCheckBox {
+                                        width: parent.width
+                                        text: model.name
+                                        checked: model.selected
+                                        visible: model.categoryId === categoryCheckbox.categoryId
+                                        onToggled: checked ? root.store.selectDiscordChannel(model.id) : root.store.unselectDiscordChannel(model.id)
```

Comment:
I'm wondering if we can do it just with one method?
```suggestion
                                        onToggled: root.store.selectDiscordChannel(model.id, checked)
```

Reply:
Made this a single method for both, channels and categories!